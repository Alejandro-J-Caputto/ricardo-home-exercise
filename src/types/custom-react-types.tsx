import { Dispatch, SetStateAction } from "react";

export type StateSetter =  Dispatch<SetStateAction<any>>