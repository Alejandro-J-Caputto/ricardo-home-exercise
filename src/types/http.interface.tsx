import {
  ArticleResponse,
  GetUserResponse,
  SearchArticle,
  SearchResponse,
} from "./response.types";

export default interface HttpConfig {
  endpoint: string;
  params: string;
  method?: string;
  headers?: Headers;
  body?: any;
}

export type cbData = (data?: GetUserResponse | ArticleResponse | SearchResponse | SearchArticle | SearchArticle[], dataIds?:string[]) => void;
