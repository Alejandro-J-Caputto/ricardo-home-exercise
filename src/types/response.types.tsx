export type User = {
  id: string;
  name: string;
};

export type GetUserResponse = User;

export type ArticleDetails = {
  id: string;
  title: string;
  subtitle: string;
  price: number;
  descriptionHtml: string;
  imageUrl: string;
  sellerId: string;
};

export type ArticleResponse = ArticleDetails;

export type SearchArticle = {
  id: number;
  title: string;
  endDate: string;
  imageUrl: string;
  buyNowPrice: number;
};

export type SearchResponse = {
  articles: SearchArticle[];
  totalCount: number;
};
