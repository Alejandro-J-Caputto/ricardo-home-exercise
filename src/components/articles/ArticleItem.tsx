import { Link } from "react-router-dom";
import { SearchArticle } from "../../types/response.types";

const ArticleItem: React.FC<{ itemsContent: SearchArticle }> = (props) => {
  const { buyNowPrice, endDate, id, imageUrl, title } = props.itemsContent;
  const dateFormat = new Date(endDate).toISOString().split("T");
  const date = dateFormat[0];
  const time = dateFormat[1];

  return (
    <div className="article" data-id={id}>
      <Link to={`/home-exercise/article/${id}`}>
        <div className="article-head">
          <img
            src={imageUrl}
            alt="test article"
            className="article-head__img"
          />
        </div>
        <div className="article-body">
          <div className="article-body__title">
            <p className="heading-article">{title}</p>
          </div>
          <div className="article-body__date">
            <p>
              <span>Ending on:</span>
              <span>{date}</span>
            </p>
            <p>
              <span>{time}</span>
            </p>
          </div>
          <div className="article-body__price">
            {buyNowPrice ? <p>{buyNowPrice} CHF</p> : <p>Keine preis</p>}
          </div>
        </div>
      </Link>
    </div>
  );
};
export default ArticleItem;
