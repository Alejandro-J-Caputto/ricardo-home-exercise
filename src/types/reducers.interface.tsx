import { ArticleDetails, SearchArticle, User } from "./response.types";

export interface ArticleInitialState {
  articles: SearchArticle[];
  articlesById: ArticleDetails;
  sellerById: User;
  savedArticlesIDs: string[];
  selectedItemsLocalStorage: SearchArticle[];
}

export interface UIinitialState {
  loadingHTTP: boolean;
  loadingApp: boolean;
  uiTheme: boolean;
}
