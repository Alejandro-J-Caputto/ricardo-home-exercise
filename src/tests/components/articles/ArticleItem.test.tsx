import { shallow, configure } from "enzyme";

import "@testing-library/jest-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import React from "react";
import { SearchArticle } from "../../../types/response.types";
import ArticleItem from "../../../components/articles/ArticleItem";

configure({ adapter: new Adapter() });
//@ts-ignore
expect.addSnapshotSerializer(createSerializer({ mode: "deep", noKey: true }));

describe("Tests on ArticlesContainer", () => {
  const article: SearchArticle = {
    id: 123,
    title: "123",
    endDate: "123",
    imageUrl: "123",
    buyNowPrice: 9000,
  };
  const wrapper = shallow(<ArticleItem itemsContent={article} />);

  test("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have the class article', () => {
    expect(wrapper.find('.article').exists()).toBe(true);
  }) 


});
