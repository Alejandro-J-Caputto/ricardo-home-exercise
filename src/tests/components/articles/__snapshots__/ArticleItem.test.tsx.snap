// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on ArticlesContainer should match the snapshot 1`] = `
<div
  className="article"
  data-id={123}
>
  <Link
    to="/home-exercise/article/123"
  >
    <div
      className="article-head"
    >
      <img
        alt="test article"
        className="article-head__img"
        src="123"
      />
    </div>
    <div
      className="article-body"
    >
      <div
        className="article-body__title"
      >
        <p
          className="heading-article"
        >
          123
        </p>
      </div>
      <div
        className="article-body__date"
      >
        <p>
          <span>
            Ending on:
          </span>
          <span>
            0122-12-31
          </span>
        </p>
        <p>
          <span>
            23:25
          </span>
        </p>
      </div>
      <div
        className="article-body__price"
      >
        <p>
          9000
           CHF
        </p>
      </div>
    </div>
  </Link>
</div>
`;
